### ISSUE TITLE
Improve warning message in `paste_linter()`

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1533]
In particular, remove the spurious `and::and()` in this message.

``` r
library(lintr)

lint(
  text = 'paste(x, collapse = ", ")',
  linters = paste_linter()
)
#> <text>:1:1: warning: [paste_linter] toString(.) is more expressive than paste(., collapse = ", "). Note also glue::glue_collapse() and and::and() for constructing human-readable / translation-friendly lists
#> paste(x, collapse = ", ")
#> ^~~~~~~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2022-09-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
why's and::and() spurious?

**Comment by IndrajeetPatil** [comment]
Yikes, sorry, my bad. 

I thought the message formatting was producing the spurious entry following the word "and"; didn't realize that there is indeed an `and::and()` function. 👀
